Sources/SwiftProtobuf/JSONEncodingVisitor.swift

```diff
@@ -30,6 +30,16 @@ internal struct JSONEncodingVisitor: Visitor {
       return encoder.stringResult
   }
 
+  /// Creates a new visitor for serializing a message of the given type to JSON
+  /// format.
+  init(type: Message.Type) throws {
+    if let nameProviding = type as? _ProtoNameProviding.Type {
+      self.nameMap = nameProviding._protobuf_nameMap
+    } else {
+      throw JSONEncodingError.missingFieldNames
+    }
+  }
+
   /// Creates a new visitor that serializes the given message to JSON format.
   init(message: Message) throws {
```

Comment:
I'm not sure if it makes sense to keep this initializer; the previous protocol extensions can call `JSONEncodingVisitor(type: Self.self)` instead, but I was trying not to touch code outside the scope of this change.